### PR TITLE
Task #156044 Fix : TjReport Bulk Email sending

### PR DESCRIPTION
**Pull request by MeghaBiranje**

### CODE REVIEW

File: tjreports/languages/administrator/en-GB/en-GB.plg_system_sendemail.ini [hunk 1]
```diff
@@ -0,0 +1,20 @@
+;@package	TECHJOOMLA
+;@copyright	Copyright © 2009-2013 TechJoomla. All rights reserved.
+;@license	GNU General Public License version 2, or later
+;@link		http://www.techjoomla.com
```
Comment (ahghatol) [inline tjreports/languages/administrator/en-GB/en-GB.plg_system_sendemail.ini]: @MeghaBiranje fix the indent

File: tjreports/plugins/system/plg_system_sendemail/bulksendemail.js [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+				jQuery("#bulkEmailModal").find("#emailsDiv").append(hiddenEle);
+			});
+
+			// Create editor field
+			 tinyMCE.execCommand( 'mceAddEditor', false, "email-message");
+
+			// alert(values.join (", "));
+		}
+		catch (err) {
+			/*console.log(err.message);*/
```
Comment (ahghatol) [inline tjreports/plugins/system/plg_system_sendemail/bulksendemail.js]: if error show on the view

**Comment by coolbung** [comment]
We seem to be adding the plugin to the tjreports repository, it should be in techjoomla library so that it can be used on any list view that shows email addresses. Also why are there no minified files @ahghatol 

cc @manojLondhe @vijaykhollam

File: tjreports/languages/administrator/en-GB/en-GB.plg_system_sendemail.ini [hunk 3]
```diff
@@ -0,0 +1,20 @@
+;@package	TECHJOOMLA
+;@copyright	Copyright © 2009-2013 TechJoomla. All rights reserved.
```
Comment (ahghatol) [inline tjreports/languages/administrator/en-GB/en-GB.plg_system_sendemail.ini]: @MeghaBiranje update Copyright year

File: tjreports/languages/administrator/en-GB/en-GB.plg_system_sendemail.ini [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+PLG_SYSTEM_SENDEMAIL_BTN="Send email"
+PLG_SYSTEM_SENDEMAIL_SELECT_CHECKBOX="Please select records."
+PLG_SYSTEM_SENDEMAIL_POPUP_HEADING="Email Notification"
+PLG_SYSTEM_SENDEMAIL_POPUP_EMAIL_SUBJECT="Subject"
+PLG_SYSTEM_SENDEMAIL_POPUP_EMAIL_BODY_MESSAGE="Message"
+PLG_SYSTEM_SENDEMAIL_POPUP_SEND_BTN="Send"
+PLG_SYSTEM_SENDEMAIL_SUCCESSFULLY_SEND="Email has been sent successfully"
+PLG_SYSTEM_SENDEMAIL_ADD_RECIPIENTS_OR_CHECK_PREFERENCES="Please add recipients or check preferences."
+PLG_SYSTEM_SENDEMAIL_ERROR="Email could not be sent."
+PLG_SYSTEM_SENDEMAIL_FALL_TO_SENDEMAIL="%s - Fail to sendemail"
```
Comment (ahghatol) [inline tjreports/languages/administrator/en-GB/en-GB.plg_system_sendemail.ini]: @MeghaBiranje  correct

send email

File: tjreports/languages/administrator/en-GB/en-GB.plg_system_sendemail.ini [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+PLG_SYSTEM_SENDEMAIL_SUCCESSFULLY_SENDEMAIL="%s - Successfully sendemail"
+
```
Comment (ahghatol) [inline tjreports/languages/administrator/en-GB/en-GB.plg_system_sendemail.ini]: correct here to send email

File: tjreports/languages/administrator/en-GB/en-GB.plg_system_sendemail.sys.ini [hunk 6]
```diff
@@ -0,0 +1,8 @@
+; @package     API
+; @subpackage  System.Tjtokenlogin
+; @copyright   Copyright (C) 2009 - 2018 Techjoomla. All rights reserved.
```
Comment (ahghatol) [inline tjreports/languages/administrator/en-GB/en-GB.plg_system_sendemail.sys.ini]: @MeghaBiranje correct year

File: tjreports/plugins/system/plg_system_sendemail/bulksendemail.js [hunk 7]
```diff
@@ -0,0 +1,282 @@
...
+				jQuery("#bulkEmailModal").find("#emailsDiv").append(hiddenEle);
+			});
+
+			// Create editor field
+			 tinyMCE.execCommand( 'mceAddEditor', false, "email-message");
+
+			// alert(values.join (", "));
+		}
+		catch (err) {
+			console.log(err.message);
```
Comment (ahghatol) [inline tjreports/plugins/system/plg_system_sendemail/bulksendemail.js]: Show alert and log error in the console under error

Comment (ahghatol) [reply tjreports/plugins/system/plg_system_sendemail/bulksendemail.js]: console.error API

File: tjreports/plugins/system/plg_system_sendemail/bulksendemail.js [hunk 8]
```diff
@@ -0,0 +1,282 @@
...
+		}
+	},
+	validate: function () {
+
+		jQuery('textarea').each(function (index, ta) {
```
Comment (ahghatol) [inline tjreports/plugins/system/plg_system_sendemail/bulksendemail.js]: @MeghaBiranje  what is the use of this block/

File: tjreports/plugins/system/plg_system_sendemail/bulksendemail.js [hunk 9]
```diff
@@ -0,0 +1,282 @@
...
+			var emailData = batch.concat(emailContent);
+
+			// Call the send email function
+			tjutilitysendemail.send(emailData, i, batchCount);
+		});
+	},
+	send: function (emailData, index, batchCount) {
+		index ++;
+
+		var url = "index.php?option=com_ajax&plugin=plg_System_Sendemail&format=json";
```
Comment (ahghatol) [inline tjreports/plugins/system/plg_system_sendemail/bulksendemail.js]: @MeghaBiranje prepend the site domain to URL. You can use Joomla.getOptions('system.paths').base

File: tjreports/plugins/system/plg_system_sendemail/bulksendemail.js [hunk 10]
```diff
@@ -0,0 +1,282 @@
...
+
+		var promise = jQuery.ajax({url: url, type: 'POST', async:true, data:emailData,dataType: 'json'});
+
+		promise.fail(
+			function(response) {
+				console.log('Something went wrong.');
```
Comment (ahghatol) [inline tjreports/plugins/system/plg_system_sendemail/bulksendemail.js]: @MeghaBiranje remove console.log

File: tjreports/plugins/system/plg_system_sendemail/bulksendemail.js [hunk 11]
```diff
@@ -0,0 +1,282 @@
...
+					jQuery('#bulkEmailModal').modal('hide');
+
+					 // Remove below line removeclass it is temp added
+					 jQuery("div").find('.tjlms-wrapper').addClass("tjBs3");
+
+					 Joomla.renderMessages({"success":[response.message]});
+				}
+			}
+		).always(
+			function(response) {
```
Comment (ahghatol) [inline tjreports/plugins/system/plg_system_sendemail/bulksendemail.js]: @MeghaBiranje remove unused block

File: tjreports/plugins/system/plg_system_sendemail/plg_system_sendemail.php [hunk 12]
```diff
@@ -0,0 +1,166 @@
+<?php
+/**
+ * @package     Plg_System_Tjlms
+ * @subpackage  Plg_System_Tjlms
+ *
+ * @author      Techjoomla <extensions@techjoomla.com>
+ * @copyright   Copyright (C) 2009 - 2018 Techjoomla. All rights reserved.
```
Comment (ahghatol) [inline tjreports/plugins/system/plg_system_sendemail/plg_system_sendemail.php]: @MeghaBiranje  correct Copyright in all files

File: tjreports/plugins/system/plg_system_sendemail/plg_system_sendemail.php [hunk 13]
```diff
@@ -0,0 +1,166 @@
...
+	}
+
+	/**
+	 * Ajax call funcation to send email
+	 *
+	 * @return  none
+	 *
+	 * @since  __DEPLOY_VERSION__
+	 */
+	public function onAjaxplg_System_Sendemail()
```
Comment (ahghatol) [inline tjreports/plugins/system/plg_system_sendemail/plg_system_sendemail.php]: @MeghaBiranje  change function name to sendEmail

File: tjreports/plugins/system/plg_system_sendemail/bulksendemail.js (hunk 7 above)
Comment (MeghaBiranje) [reply tjreports/plugins/system/plg_system_sendemail/bulksendemail.js]: Sir I didn't understand console.errorAPI so only gave alert message now.

**Comment by twsvaishali** [comment]
@coolbung Instead of pushing this to tj library, we should have new repo like assets loader etc

**Comment by coolbung** [comment]
Yes, see my comment prior to yours. Please go ahead and do that @twsvaishali

**Comment by twsvaishali** [comment]
Yes.. already in process..

**Comment by twsvaishali** [comment]
MR : https://github.com/techjoomla/plg_system_sendemail/pull/4/files

**Comment by twsvaishali** [comment]
MR for bulksend email plugin:
https://github.com/techjoomla/plg_system_sendemail/pull/4/files
MR for Tjreports: https://github.com/techjoomla/com_tjreports/pull/179/files
MR for Report plugin to support email column :
https://git.tekdi.net/techjoomla/shika/merge_requests/4250

**Pull request closed without merging**